internal/chain/beacon/cache.go

```diff
@@ -38,17 +39,22 @@ func roundID(round uint64, previous []byte) string {
 }
 
 // Append adds a partial signature to the cache.
-func (c *partialCache) Append(p *drand.PartialBeaconPacket) {
+func (c *partialCache) Append(p *drand.PartialBeaconPacket) error {
 	id := roundID(p.GetRound(), p.GetPreviousSignature())
-	idx, _ := c.scheme.ThresholdScheme.IndexOf(p.GetPartialSig())
-	round := c.getCache(id, p)
-	if round == nil {
-		return
+	idx, err := c.scheme.ThresholdScheme.IndexOf(p.GetPartialSig())
+	if err != nil {
+		c.l.Errorw("partialCache could not get index of threshold scheme", "err", err)
```

Comment:
Do we want to add the name of the threshold scheme used (in the error message logged) or not necessary? @AnomalRoil

Reply:
No, we're using the same threshold scheme everywhere atm.